common/script/ops/openMysteryItem.js

```diff
@@ -1,32 +1,41 @@
 import content from '../content/index';
+import i18n from '../i18n';
+import {
+  BadRequest,
+} from '../libs/errors';
+
+module.exports = function openMysteryItem (user, req = {}, analytics) {
+  let item = user.purchased.plan.mysteryItems.shift();
 
-module.exports = function(user, req, cb, analytics) {
-  var analyticsData, item, ref, ref1;
-  item = (ref = user.purchased.plan) != null ? (ref1 = ref.mysteryItems) != null ? ref1.shift() : void 0 : void 0;
   if (!item) {
-    return typeof cb === "function" ? cb({
-      code: 400,
-      message: "Empty"
-    }) : void 0;
+    throw new BadRequest(i18n.t('mysteryItemIsEmpty', req.language));
   }
+
   item = content.gear.flat[item];
+  item.notificationType = 'Mystery';
   user.items.gear.owned[item.key] = true;
-  if (typeof user.markModified === "function") {
+
+  if (typeof user.markModified === 'function') {
```

Comment:
The original line was there to only run the markmodified code on the server. To cut down on unnecessary if statements, we're going to add a noop markmodified function to the client user object. So you don't need the if statement.